reservation/reservation.go

```diff
@@ -106,6 +106,18 @@ func (manager *Manager) Lock(resource string) (*Reservation, error) {
 	return res, nil
 }
 
+// WaitUntilLock creates a Reservation for `resource`, or waits until it can do so.
+func (manager *Manager) WaitUntilLock(resource string) (*Reservation, error) {
+	res, err := manager.Lock(resource)
+	for fmt.Sprintf("%s", err) == fmt.Sprintf("Reservation already exists for resource %s", resource) {
```

Comment:
Can you still move this out into its own function? I think it would make this condition a lot easier to understand. You could put it at the top level and have it take in `err` and `resource`, or put it in `WaitUntilLock` and have it only take in `err`, e.g.

```go
reservationAlreadyExists := func(err error) {
  return fmt.Sprintf("%s", err) == fmt.Sprintf("Reservation already exists for resource %s", resource)
}
```